core/src/blocktree_processor.rs

```diff
@@ -616,6 +617,72 @@ mod tests {
         assert_eq!(bank.get_balance(&keypair3.pubkey()), 2);
     }
 
+    #[test]
+    fn test_process_entries_2_txes_collision_and_error() {
+        let (genesis_block, mint_keypair) = GenesisBlock::new(1000);
+        let bank = Bank::new(&genesis_block);
+        let keypair1 = Keypair::new();
+        let keypair2 = Keypair::new();
+        let keypair3 = Keypair::new();
+
+        // fund: put 4 in each of 1 and 2
+        assert_matches!(bank.transfer(4, &mint_keypair, &keypair1.pubkey()), Ok(_));
+        assert_matches!(bank.transfer(4, &mint_keypair, &keypair2.pubkey()), Ok(_));
+
+        // construct an Entry whose 2nd transaction would cause a lock conflict with previous entry
+        let entry_1_to_mint = next_entry(
+            &bank.last_blockhash(),
+            1,
+            vec![SystemTransaction::new_account(
+                &keypair1,
+                &mint_keypair.pubkey(),
+                1001, // Should cause a transaction failure
```

Comment:
@garious, I'm seeing SystemProgram throw an error (insufficient funds) in my logs, but it's not propagating through to par_process_entries(). I remember you mentioning something about bubbling up errors from programs a while back, do you have any ideas off the top of your head?

Reply:
I think you have the arguments backwards. Afk so I could be wrong..

Reply:
Could it be due to this Err->Ok that I've been complaining about? https://github.com/solana-labs/solana/blob/master/runtime/src/bank.rs#L696

Reply:
@garious, yup, its that :P

Reply:
You'd need to ask @pgarg66. I think Ok at this layer changes meaning from "transaction processed as expected" to "bank took a transaction fee".  Maybe it made more sense when the error was called BankError. Now that it's called TransactionError, it's quite confusing.